src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RoleSyncService.kt

```diff
@@ -71,21 +72,22 @@ class RoleSyncService(
       telemetryClient.trackEvent("HMUA-Role-Change-Failure", mapOf("roleCode" to currentRoleData.roleCode), null)
     } else if (!diff.areEqual()) {
       stats.roles[newRoleData!!.roleCode] = RoleDifferences(newRoleData.roleCode, diff.toString())
-      try {
-        storeInNomis(currentRoleData, newRoleData, stats)
-        if (stats.roles[newRoleData.roleCode]?.updateType != NONE) {
-          val trackingAttributes = mapOf(
-            "roleCode" to newRoleData.roleCode,
-            "differences" to stats.roles[newRoleData.roleCode]?.differences,
-          )
-          telemetryClient.trackEvent("HMUA-Role-Change", trackingAttributes, null)
+      if (!readOnly)
```

Comment:
struggling to understand the code here - if it is read only do we then report or anything on the differences?

Also maybe clearer to do something like
```
if (readOnly) return
```
and then the rest of the code doesn't need to be indented etc.